octodns/provider/ns1.py

```diff
@@ -646,6 +652,10 @@ def _data_for_CAA(self, _type, record):
         }
 
     def _data_for_CNAME(self, _type, record):
+        if record.get('tier', 1) > 1:
+            # Advanced dynamic record
+            return self._data_for_dynamic_A(_type, record)
```

Comment:
We should probably go ahead and rename this method now that it supports A, AAAA,  and CNAME. `_data_for_dynamic` is probably sufficient. (It already did A and AAAA so the name wasn't great...)

Reply:
Went ahead and renamed both _data_for_dynamic_A and _params_for_dynamic_A methods.